### DISCUSSION TITLE
Edit and deploy a dapp in remix

### DISCUSSION THREAD

**Discussion by drafish** [PlanckerLabs/remix-cc#109] [Ideas]
## Edit dapp
![edit-dapp@1x](https://github.com/PlanckerLabs/remix-cc/assets/18543527/8ee047ce-aea2-4d66-9747-cb10b7b818d5)

There is a button named `Edit` in each contract instance. Clicking the `Edit` button will open a new tab in the main panel where you can edit the dapp. 

Initially, it only has one group which contains all functions. You can add groups by clicking the `ADD PAGE` button.  Then you can drag and drop any functions into any groups you want. Clicking the `RESET` button will reset it to the initial state.

Once you're done with editing, you can deploy it to `surge.sh`. First you need to input your email and password for your surge account, as well as a unique sub-domain. Then click the `DEPLOY` button. Once the dapp site is deployed, you will receive a url like this `https://some-dapp.surge.sh`. You can copy and open it in a new tab.

The static files of all dapp sites are almost the same. The only difference between them is the json which contains the information of the dapp, such as address, abi, network, etc. Clicking the `RELOAD` button will load the json from surge, and set the editing state to the last time you deployed it.

## View dapp
![view-dapp@1x](https://github.com/PlanckerLabs/remix-cc/assets/18543527/7876548a-78cf-407a-899f-fdb3473b4bde)

Check the image above. That's how the dapp site looks like. First you need to connect to the wallet. Then you can interact with these functions just like you do in remix. 

Each group has its own page. It defaults to show functions in home group. If you want to interact with other functions, just go to other pages.

**Comment by ryestew** [comment]
I'm not sure you need more than 1 page in the main panel view -
<img width="331" alt="image" src="https://github.com/PlanckerLabs/remix-cc/assets/6050001/840e5173-37c1-4dec-8359-49d1cf94d36b">

Why would you need more than 1 page? (There might be a good reason...)

Instead of drag and drop the functions, maybe a simpler solution (although not as elegant - but maybe easier of the first iteration) would to use a select box to choose a function for including in the published DApp.   But on the other hand a checkbox would not let you order the functions and that is really important.  

The main panel for this 1ClickDapp could be called the Staging Area before it is published to surge.

Do you think it would be interesting to have some text boxes between the functions - so that there could be some instructions for someone trying out the DApp?

**Reply by drafish** [reply]
>Why would you need more than 1 page?

I'm not sure either. I thought someone may want multi dapps based on one contract. Function 1-5 will be dapp one, 6-10 will be dapp two, something like that. And it will be a waste to depoy them in different sub-domains, pages will be more suitable.

Anyway, we can start with 1 page, and add more pages in the further iterations if users really need it.

>use a select box to choose a function for including in the published DApp

You mean checkbox, right? Sure it will be an easier way.

>Do you think it would be interesting to have some text boxes between the functions

Yeah, that's a great idea.
